### ISSUE TITLE
Idea - Follow implicit links?

### ISSUE THREAD

**Issue by trev-dev** [jakewvincent/mkdnflow.nvim#37]
I am feeling picky about not having to replicate a bunch of notes that will also belong to some static site generator that cannot drop the "markdown" extension and just build a valid route.

I am also noticing that GitJournal/ZettleNotes can happily follow implicit links on Android.

This is strictly a "have your cake and eat it too" scenario, but maybe if this were an optional feature in the config such as "followImplicitLinks = false" by default. When true, links such as `[I am a link](some-file)` could automatically resolve to `some-file.md` or whatever other possible format we've chosen to accept as flowy in the `filetypes` table will do.

**Comment by jakewvincent** [comment]
This is reasonable! My current thought is to create a `links` table that will contain settings for both the link style (currently its own config option) and implicit extension:
```lua
require('mkdnflow').setup({
    links = {
        style = 'markdown',
        implicit_extension = 'md'
    }
})
```
I think rather than disabling interpretation of implicit extensions by default, it would be nice for the default behavior to be flexible--if there's no extension, the link is interpreted with an implicit extension... if there is, that's taken to be an explicit extension without issue. I'll try this out right now.

**Comment by jakewvincent** [comment]
I guess even if interpretation is flexible, there's the question of whether links should be created with or without extensions by default. That should probably be a config option. I'm not sure what most static site generators expect, but I believe Jekyll at least expects an explicit `.md` extension. Which static site generator are you using? Any idea what's most common?

**Comment by trev-dev** [comment]
I'm using Hugo, and the question of how to get its parser to drop the .md had been asked a few times and has gone unanswered. I had considered some other option such as pelican but I wanted to use something closer to GoLang because learning.

**Comment by jakewvincent** [comment]
or sorry, I meant any idea which behavior is most common when it comes to static site generators? Do you know if most tend not to expect an extension?

(I almost have this ready for testing, btw)

**Comment by jakewvincent** [comment]
@trev-dev can you try this out on the dev branch and see if this is working as expected? If you're using Packer, you'd do the following. You'll need to stipulate an implicit extension in order to get this working, so assuming you're working with markdown files, you'd do:
```lua
use({
    'jakewvincent/mkdnflow.nvim',
    branch = 'dev',
    config = function()
        require('mkdnflow').setup({
            -- ...
            links = {
                implicit_extension = 'md'
            }
            -- ...
        })
    end
})
```

**Comment by trev-dev** [comment]
> or sorry, I meant any idea which behavior is most common when it comes to static site generators? Do you know if most tend not to expect an extension?
> 
> (I almost have this ready for testing, btw)

 I have only used Pelican & Hugo in terms of static site generators that are not just my fingers and a text editor. I assume (bad!) that the expected behavior is to link to the expected generated route somehow, be it explicitly or via some route generator filter/method/function.
 
Hugo is interesting in that I may make _relative_ links to content. As such, the extension itself is the only thing standing in my way...and I won't have to double-up on work or give up working links in one place or another.

> can you try this out on the dev branch and see if this is working as expected?

_Yes!_

**Comment by jakewvincent** [comment]
[Note for self] Anchors within a link's source are currently only interpreted correctly if the .md extension is explicit. e.g. the following attempts to open a file `document.mdanchor.md` instead of opening `document.md` and trying to find a heading matching `# Anchor` within that document.

```markdown
[Link name](document#anchor)
```

**Comment by jakewvincent** [comment]
Addressed above comment in commit 0c305cb (currently in dev branch)
